src/node/node_class.hpp

```diff
@@ -37,256 +57,1269 @@ using IndexPropertyType = js::IndexPropertyType<Types>;
+	Napi::Symbol externalSymbol = ExternalSymbol;
+	//if we are creating a RealmObject from schema with no user defined constructor
+	if (constructor.IsEmpty()) {
+		//1.Check by name if the constructor is already created for this RealmObject 
+		if (!schemaObjects->count(schemaName)) {
+
+			//2.Create the constructor
+
+			//get or create the RealmObjectClass<T> constructor
+
+			//create the RealmObject function by name
+			schemaObjectConstructor = Napi::Function::New(env, schema_object_type_constructor, schema.name);
+
+			Napi::Function realmObjectClassConstructor = ObjectWrap<ClassType>::create_constructor(env);
+			auto parentCtorPrototypeProperty = realmObjectClassConstructor.Get("prototype");
+			auto childPrototypeProperty = schemaObjectConstructor.Get("prototype").As<Napi::Object>();
+			ObjectSetPrototypeOf.Call({ childPrototypeProperty, parentCtorPrototypeProperty });
+			ObjectSetPrototypeOf.Call({ schemaObjectConstructor, realmObjectClassConstructor });
+
+			//get all properties from the schema 
+			std::vector<Napi::PropertyDescriptor> properties;
+			for (auto& property : schema.persisted_properties) {
+				std::string propName = property.public_name.empty() ? property.name : property.public_name;
+				node::String* name = getCachedPropertyName(propName);
+				auto descriptor = Napi::PropertyDescriptor::Accessor<property_getter, property_setter>(name->ToString(env), (napi_property_attributes)(napi_writable | napi_enumerable), 
+					(void*)name);
+				properties.push_back(descriptor);
+			}
+
+			for (auto& property : schema.computed_properties) {
+				std::string propName = property.public_name.empty() ? property.name : property.public_name;
+				node::String* name = getCachedPropertyName(propName);
+				auto descriptor = Napi::PropertyDescriptor::Accessor<property_getter, property_setter>(name->ToString(env), (napi_property_attributes)(napi_writable | napi_enumerable),
+					(void*)name);
+				properties.push_back(descriptor);
+			}
+
+			//define the properties on the prototype of the schema object constructor
+			childPrototypeProperty.DefineProperties(properties);
+			
+			SchemaObjectType* schemaObjectType = new SchemaObjectType();
+			schemaObjects->emplace(schemaName, schemaObjectType);
+			schemaObjectType->constructor = Napi::Persistent(schemaObjectConstructor);
+			schemaObjectType->constructor.SuppressDestruct();
+		}
+		else {
+			//hot path. The constructor for this schema object is already cached. use it and return a new instance
+			SchemaObjectType* schemaObjectType = schemaObjects->at(schemaName);
+			schemaObjectConstructor = schemaObjectType->constructor.Value();
+		}
+
+		Napi::External<Internal> externalValue = Napi::External<Internal>::New(env, internal, internalFinalizer);
+		instance = schemaObjectConstructor.New({});
+		instance.Set(externalSymbol, externalValue);
+		
+		Napi::Object proto = GetPrototype(env, instance);
+		proto.Set(externalSymbol, externalValue);
+	}
+	else {
```

Comment:
Could we reuse a bit more between these two code paths? It seems to me the generation of property descriptors are very similar across the two.